node/cn/handler.go

```diff
@@ -522,8 +523,9 @@ func (pm *ProtocolManager) processConsensusMsg(msgCh <-chan p2p.Msg, p Peer, add
 		if handler, ok := pm.engine.(consensus.Handler); ok {
 			_, err := handler.HandleMsg(addr, msg)
 			// if msg is istanbul msg, handled is true and err is nil if handle msg is successful.
-			if err != nil {
+			if err == istanbul.ErrStoppedEngine {
```

Comment:
Why don't you send an error to `errCh` for this?

Reply:
This error is printed at here when shutting down the istanbul engine, and it is lead to the disconnection. I think disconnection is not proper. The reason is because the shutdown of an istanbul engine is caused by sync. During sync, it still needs connection. Rather, the two peer is disconnected.

As always, my thought can be wrong, so if you have any concern, i can revert back to disconnect the connection.

Reply:
In the currency full mesh CN network, your opinion is right. I am a little bit afraid of unexpected situations, but we can evaluate them in the QA process. 
Thank you!